src/WebJobs.Script/FileTraceWriter.cs

```diff
@@ -38,9 +40,17 @@ public FileTraceWriter(string logFilePath, TraceLevel level) : base(level)
             }
             else
             {
```

Comment:
Conceivably, a host instance could wake up with lots of pending requests and end up executing this code multiple times. But since you ignore delete exceptions, it should not blow up.